src/logic.js

```diff
@@ -237,6 +240,12 @@ define([
             _.each(expr.instanceRefs, function (ignore, id) {
                 form.referenceInstance(id, mug, property);
             });
+            _.each(expr.hashtags, function (hashtag) {
+                if (/^#case/.test(hashtag.toHashtag())) {
+                    form.referenceInstance('casedb', mug, property);
+                    form.referenceInstance('commcaresession', mug, property);
```

Comment:
I think we do not want to make the logic manager have to know which hashtags reference which instances. It doesn't seem like the right place for it because of the hard coded `#case` and `casedb` and `commcaresession`, which are all form properties. What about moving this into the form with an API like `form.referenceHashtag(hashtag, mug, property)`? The form knows about hashtag namespaces like `#form` and `#case`, and it knows about instances, so adding the mapping between them there seems better than here.